hadoop-ozone/integration-test/src/test/java/org/apache/hadoop/ozone/om/multitenant/RangerUserRequest.java

```diff
@@ -0,0 +1,275 @@
+import static org.apache.hadoop.ozone.OzoneConsts.OZONE_OM_RANGER_ADMIN_GET_USER_HTTP_ENDPOINT;
+
+/**
+ * Helper class to create and delete users in Ranger because RangerClient
+ * doesn't support it yet.
+ */
+public class RangerUserRequest {
+
+  private static final Logger LOG =
+      LoggerFactory.getLogger(RangerUserRequest.class);
+
+  private String rangerEndpoint;
+
+  // Value of HTTP auth header
+  private String authHeaderValue;
+
+  RangerUserRequest(String rangerHttpsAddress, String userName, String passwd) {
+
+    // Trim trailing slash
+    if (rangerHttpsAddress.endsWith("/")) {
+      rangerHttpsAddress =
+          rangerHttpsAddress.substring(0, rangerHttpsAddress.length() - 1);
+    }
+    this.rangerEndpoint = rangerHttpsAddress;
+
+    String auth = userName + ":" + passwd;
+    byte[] encodedAuth =
+        Base64.encodeBase64(auth.getBytes(StandardCharsets.UTF_8));
+    authHeaderValue = "Basic " +
+        new String(encodedAuth, StandardCharsets.UTF_8);
+
+    setupRangerIgnoreServerCertificate();
+  }
+
+  private void setupRangerIgnoreServerCertificate() {
+    // Create a trust manager that does not validate certificate chains
+    TrustManager[] trustAllCerts = new TrustManager[]{
+        new X509TrustManager() {
+          public java.security.cert.X509Certificate[] getAcceptedIssuers() {
+            return null;
+          }
+          public void checkClientTrusted(
+              java.security.cert.X509Certificate[] certs, String authType) {
+          }
+          public void checkServerTrusted(
+              java.security.cert.X509Certificate[] certs, String authType) {
+          }
+        }
+    };
+
+    try {
+      SSLContext sc = SSLContext.getInstance("SSL");
+      sc.init(null, trustAllCerts, new java.security.SecureRandom());
+      HttpsURLConnection.setDefaultSSLSocketFactory(sc.getSocketFactory());
+    } catch (Exception e) {
+      LOG.info("Setting DefaultSSLSocketFactory failed.");
+    }
+  }
+
+  private HttpURLConnection openURLConnection(URL url) throws IOException {
+    final HttpURLConnection urlConnection;
+    if (url.getProtocol().equals("https")) {
+      urlConnection = (HttpsURLConnection) url.openConnection();
+    } else if (url.getProtocol().equals("http")) {
+      urlConnection = (HttpURLConnection) url.openConnection();
+    } else {
+      throw new IOException("Unsupported protocol: " + url.getProtocol() +
+          "URL: " + url);
+    }
+    return urlConnection;
+  }
+
+  /**
+   * Can make either http or https request.
+   */
+  private HttpURLConnection makeHttpCall(String urlString,
+      String jsonInputString, String method, boolean isSpnego)
+      throws IOException {
+
+    URL url = new URL(urlString);
+    final HttpURLConnection urlConnection = openURLConnection(url);
+
+    urlConnection.setRequestMethod(method);
+    // Timeout in ms
+    urlConnection.setConnectTimeout(5000);
```

Comment:
ToDo: Make timeout values configurable

Reply:
Yes. The actual production timeout is tunable: https://github.com/apache/ozone/blob/fc3a23af25d61146139ad434b2323bbeedc89828/hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/multitenant/RangerRestMultiTenantAccessController.java#L440

`RangerUserRequest` is test-only code so I didn't bother. I could read the some config as well though.

Reply:
Added in 72a4d60